### ISSUE TITLE
Error: inputSpec or inputSpecRootDirectory must be specified

### ISSUE THREAD

**Issue by Vtfelker** [OpenAPITools/openapi-generator#19523]
I added the plugin like this:
`<plugin>
                <groupId>org.openapitools</groupId>
                <artifactId>openapi-generator-maven-plugin</artifactId>
                
                <version>7.8.0</version>
                
                <executions>
                    <execution>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                        <configuration>
                            <inputSpec>${project.basedir}/vbss-rest/src/main/resources/swagger.yaml</inputSpec>
                            <generatorName>java</generatorName>
                            <configOptions>
                                <sourceFolder>src/gen/java/main</sourceFolder>
                            </configOptions>
                        </configuration>
                    </execution>
                </executions>
            </plugin>`

But I get the following error: "inputSpec or inputSpecRootDirectory must be specified". Why?

**Comment by alla-gofman** [comment]
Related:
https://stackoverflow.com/questions/65609405/how-do-i-fix-openapi-error-in-seemingly-correct-inputspec-for-code-generation

https://github.com/OpenAPITools/openapi-generator/issues/2207

**Comment by MarcellHarmaci** [comment]
I ran into the same issue. Didn't manage to solve it yet.

Here's a human readable formatting of @Vtfelker's plugin configuration:
```
<plugin>
    <groupId>org.openapitools</groupId>
    <artifactId>openapi-generator-maven-plugin</artifactId> 
    <!-- RELEASE_VERSION -->
    <version>7.8.0</version> 
    <!-- /RELEASE_VERSION -->
    <executions>
        <execution>
            <goals>
                <goal>generate</goal>
            </goals>
            <configuration>
                <inputSpec>${project.basedir}/vbss-rest/src/main/resources/swagger.yaml</inputSpec>
                <generatorName>java</generatorName>
                <configOptions>
                    <sourceFolder>src/gen/java/main</sourceFolder>
                </configOptions>
            </configuration>
        </execution>
    </executions>
</plugin>
```

Looks like it's the same as the one presented in [https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-maven-plugin](https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-maven-plugin)

**Comment by vitran96** [comment]
## TLDR

This plugin cannot be called directly with `mvn openapi-generator:generate`

The following commands will generate the code:
* `mvn generate-sources`
* `mvn compile`

**Comment by HK-hub** [comment]
I use these command get other Error:  I used JDK 21

![Image](https://github.com/user-attachments/assets/3b33cf56-bb11-4220-9c31-69362999e48e)
